elobot.py

```diff
@@ -54,245 +30,233 @@ def get_channel_id(self, channel_name):
+            new_elo = get_elo(handle)
+            self.elo_cache[handle] = new_elo
+            if verbose:
+                elo_delta = new_elo - old_elo
+                if elo_delta != 0:
+                    self.talk_to(handle, f'Your ELO is {new_elo} ({show(elo_delta)})')
 
     def talk(self, message):
         """Send a message to the Slack channel"""
         self.slack_client.api_call('chat.postMessage', channel=self.channel_id, text=message, username=self.name)
 
-    def talk_to(self, user_id, message):
+    def talk_to(self, handle_s, message):
+        """Accepts a single handle or a list of handles."""
         message = message[0].lower() + message[1:]
-        self.talk(f'<@{user_id}>, {message}')
+
+        if isinstance(handle_s, list) or isinstance(handle_s, set):
+            self.talk(colloq_listify(f'<@{handle}>' for handle in set(handle_s)) + ': ' + message)
+        else:
+            self.talk(f'<@{handle_s}>, {message}')
 
     def run(self):
+        self.slack_client.rtm_connect(auto_reconnect=True)
         while True:
             time.sleep(0.1)
-            self.heartbeat()
-            self.ensure_connected()
-
             messages = self.slack_client.rtm_read()
             for message in messages:
-                if 'user' in message and message.get('type', False) == 'message' and message.get('channel', False) == self.channel_id and message.get('text', False):
+                if 'user' in message and 'text' in message and message.get('type') == 'message' and message.get('channel') == self.channel_id:
                     self.handle_message(message)
 
     def handle_message(self, message):
-        print(f'Message received:\n{message}')
-
         text = message['text']
         user_handle = message['user']
 
-        if BACKDOOR_ENABLED and BACKDOOR_REGEX.match(text):
-            new_user_handle, new_text = re.search(BACKDOOR_REGEX, text).groups()
+        if BACKDOOR_ENABLED and re.match(BACKDOOR_REGEX_G, text, re.IGNORECASE):
+            new_user_handle, new_text = re.match(BACKDOOR_REGEX_G, text, re.IGNORECASE).groups()
             return self.handle_message({
                 'user': new_user_handle,
                 'text': new_text
             })
 
-        if WINNER_REGEX.match(text):
-            loser_handle, winner_score, loser_score = re.search(WINNER_REGEX, text).groups()
-            self.winner(user_handle, loser_handle, int(winner_score), int(loser_score))
-        elif CONFIRM_REGEX.match(text):
-            match_id, = re.search(CONFIRM_REGEX, text).groups()
-            self.confirm(user_handle, match_id)
-        elif CONFIRM_ALL_REGEX.match(text):
+        if re.match(GAME_REGEX_G, text, re.IGNORECASE):
+            winner_handles, loser_handles, scores = parse_game(text, user_handle)
+
+            # Ensure that no player is in the game more than once
+            # This is only a porcelain restriction; the code otherwise allows repeat players
```

Comment:
the hell is a porcelain restriction?

Reply:
I stole the term "porcelain" from git to mean "interfacing/dealing with/relating to" a human. In this case, I mean that it's a design choice, not a technical limitation.

Reply:
dang that's terminology I've never heard before